### ISSUE TITLE
Adjust ArticlePolicy to incorporate conditional logic of FeatureFlag

### ISSUE THREAD

**Issue by jeremyf** [forem/forem#16483]
This builds on the work of #16482

**Comment by jeremyf** [comment]
@amywtlin now comes the big question:  When we say "admin" which all admins do we mean?

In DEV we have the following named roles:

| Name | Recommendation |
|-|-|
| admin | Yes |
| codeland_admin | No? |
| mod_relations_admin | No? |
| podcast_admin | No! |
| single_resource_admin | Hard No! |
| super_admin | Yes |
| support_admin | No? |
| tech_admin | No? |

Optionally, we could configure a setting to say "What roles must someone have to create an article?"

**Comment by amywtlin** [comment]
I am glad you asked. Here's what I think (and it's up for debate, as always):

1. Super admins should be the ones who can turn on/off the comment-mode. The rationale being, It's a config affecting the whole site, and the current authorization rule is that only super admins can touch such configurations under `/config` (see more in [admin roles](https://docs.google.com/document/d/1_MUj3IsbK1g5Cf2NyUk_MAoz0SIYzROLQ8XNdsCPfrE/edit#heading=h.dc4dw8udn6yf)). Though we have decided to put the switch under a new page that is not `/config`, it still shares the same nature of configurations under `/config`. 
2. Both admins & super admins can publish articles. 
3. The rest of the types of admins are highly customized for DEV, such as tech admins & other single-resource admins (see the same doc about admin roles on "General Rules"), hence I don't think they'll exist in other Forems. In any case they happen to exist in other Forems, they shouldn't be able to publish posts.

To sum up, the table would go: 

Name | Recommendation
-- | --
admin | Can post
codeland_admin | No!
mod_relations_admin | No!
podcast_admin | No!
single_resource_admin | Hard No!
super_admin | Can flip the switch & post
support_admin | No!
tech_admin | No!
